test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -8817,6 +8861,40 @@ TEST_F(AtenXlaTensorTest, TestAvgPool3DNoBatchBackward) {
   }
 }
 
+TEST_F(AtenXlaTensorTest, TestAdaptiveAvgPool3DNoBatchBackward) {
+  for (int64_t output_size : {7, 8}) {
+    auto testfn =
+        [&](const std::vector<torch::Tensor>& inputs) -> torch::Tensor {
+      return torch::adaptive_avg_pool3d(
+          inputs[0], {output_size, output_size, output_size});
+    };
+    ForEachDevice([&](const torch::Device& device) {
+      TestBackward(
+          {torch::rand(
+              {1, 32, 28, 28},
+              torch::TensorOptions(torch::kFloat).requires_grad(true))},
+          device, testfn);
+    });
+  }
+}
```

Comment:
Can you add counter checks like
```
  ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
  ExpectCounterChanged("xla::adaptive_avg_pool3d", cpp_test::GetIgnoredCounters());
```
to make sure xla version of ops is run?

Reply:
Since the IsSupported function returns false, when requested input dimensions are not divisible by requested output dimension. Counter tests will pass only if that condition is met. Should I update the test or leave out this check. For adaptive_avg_pool2d this check was not implemented, so I followed the example. Please advise, I can add the checks for both 2d and 3d ops but the output and input sizes will need to be modified accordingly.

Reply:
You can only check `  ExpectCounterChanged("xla::adaptive_avg_pool3d", cpp_test::GetIgnoredCounters());` then, xla counter will be increased regardless of the input.

Reply:
I made the changes to the output input shapes so both checks work. Is it ok or I revert and add only the ExpectCounterChanged ?

Reply:
It is ok to check both.

Reply:
Ok, just pushed the changes.